docs/sources/gateway-api.md

```diff
@@ -202,6 +202,8 @@ spec:
         - --source=gateway-tlsroute
         - --source=gateway-tcproute
         - --source=gateway-udproute
+        # Optionally, enable Gateway API Experimental resource support (XListenerSet)
+        - --gateway-enable-experimental
```

Comment:
I understand it's an experimental feature.
It may become stable one day, though. 

Enabling it with `--source=gateway-listener` instead of introducing this different on/off switch would be easier for users. What should motivate us to introduce this inconsistency?

Reply:
I'm not tied to the additional flag in this case.

The thought process at the time was a `ListenerSet` isn't a first-class source in the same way a `Gateway` is not. Rather a method of finding the source address for the attached route. 

I will change it to an additional source as you suggested.

Reply:
Ok, having re-reviewed this adding it as an additional source doesn't really fit with how the `--source`'s are instantiated as there's no additional source to create. It's a switch on how the existing `gateway-*route` sources function.

The traefik source has something similar in that there are `--traefik-enable-legacy` and `--traefik-disable-new` options.